### PR TITLE
fix(remix-dev): support multiple nested pathless layout routes

### PR DESCRIPTION
**Pull request by pokedotdev**
Closes: #3327

- [ ] Docs
- [x] Tests

### CODE REVIEW

File: packages/remix-dev/config/routesConvention.ts [hunk 1]
```diff
@@ -62,13 +62,16 @@ export function defineConventionalRoutes(
...
 
-      let isIndexRoute = routeId.endsWith("/index");
+      let isIndexRoute = lastSegment === "index";
+      let isLayoutRoute = lastSegment.startsWith("__");
       let fullPath = createRoutePath(routeId.slice("routes".length + 1));
-      let uniqueRouteId = (fullPath || "") + (isIndexRoute ? "?index" : "");
+      let uniqueRouteId = fullPath || "";
+
+      if (isIndexRoute) uniqueRouteId += "?index";
+      else if (isLayoutRoute) uniqueRouteId += lastSegment;
```
Comment (MichaelDeBoey) [inline packages/remix-dev/config/routesConvention.ts:74]: Let's make this less error-prone

```suggestion
      if (isIndexRoute) {
        uniqueRouteId += "?index";
      } else if (isLayoutRoute) {
        uniqueRouteId += lastSegment;
      }
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Comment by apalumbo** [comment]
@MichaelDeBoey is this PR still to approve?

**Comment by brophdawg11** [comment]
Oh interesting, I didn't know we had a separate PR for this. This is also fixed by #4421 which was initially addressing #3014 but uncovered this same issue during testing.  I think I lean towards the solution in #4421 since rather than mess with the `id` just to make it unique, we skip the check entirely for pathless layout routes which, by definition, create the potential for path collisions.  I'll check with the team as well and we'll get one of these merged to fix #3327.  

It may make sense to keep the integration tests from this PR even if we go with #4421 - I'll take a closer look and make sure we're properly covered.

File: integration/multiple-pathless-routes-test.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+});
+
+test.afterAll(async () => appFixture.close());
+
+test("slug index & foo layout", async () => {
+  let res = await fixture.requestDocument("/slug");
+  let text = await res.text();
+  expect(text).toMatch(TEXTS.SLUG_LAYOUT);
+  expect(text).toMatch(TEXTS.FOO_LAYOUT);
+  expect(text).toMatch(TEXTS.SLUG_LAYOUT);
```
Comment (Vivalldi) [inline integration/multiple-pathless-routes-test.ts:76]:
```suggestion
  expect(text).toMatch(TEXTS.SLUG_CHILD);
```

**Review by Vivalldi — COMMENTED** [review]
Noticed a small thing regarding the `slug index & foo layout` test

**Comment by chaance** [comment]
This got a bit outdated -- very sorry it took us a bit to follow up. As @brophdawg11 mentioned this should have been taken care of already, and most of the logic has been moved to React Router. If there is still an issue I'd suggest opening one in that repo so we can give it a fresh look.

**Pull request closed without merging**